modules/streaming/src/main/scala/io/circe/streaming/ParsingConfiguration.scala

```diff
@@ -0,0 +1,11 @@
+package io.circe.streaming
+
+import jawn.AsyncParser
+
+case class ParsingConfiguration(parseMode: AsyncParser.Mode)
+
+object ParsingConfiguration {
+
+  implicit val defaultParsingConfiguration: ParsingConfiguration =
```

Comment:
I'm not a huge fan of implicit parameters used for configuration. I'd rather prefer us to be explicit about this. One way of doing this is to encode the parsing mode in the function name (e.g., `stringStreamParser` and `byteStreamParser` - maybe there are better names).

An alternative solution would be to add a new parsing mode to Jawn (something like `Smart` mode) when it tries to use `StreamValue` by default and then fall back to `UnwrapArray` if the top-level object is an array.